backend/oasst_backend/prompt_repository.py

```diff
@@ -344,6 +344,26 @@ def insert_toxicity(self, message_id: UUID, model: str, score: float, label: str
         self.db.add(message_toxicity)
         return message_toxicity
 
+    def check_users_recent_replies_for_duplicates(self, task_interaction: protocol_schema.AnyInteraction) -> bool:
+        """Check if the user has a task of the same type with the same text in the last 24 hours."""
+        username = task_interaction.user.id
+        user = self.db.query(User).filter(User.username == username).first()
```

Comment:
no need to query, the user is available in self.user_id (btw. username is not unique.. only the combination of username + auth_method, see `lookup_client_user()`)